src/modules/api/component.ts

```diff
@@ -223,8 +266,29 @@ export async function createApiComponent(components: {
         }
       }`
       )
+      const updatedRentalListings =
+        rentals.getUpdatedRentalListings(updatedAfter)
+
+      // Gets the latest parcels, estates and rental listings.
+      const [{ parcels, estates }, rentalListings] = await Promise.all([
+        updatedLand,
+        updatedRentalListings,
+      ])
+
+      // Gets the rental listings by nft id to use them more efficiently later.
+      const rentalListingByNftId = rentalListings.reduce(
+        (prev, current) => ({
+          ...prev,
```

Comment:
Try to avoid looping within a reduce, the complexity grows exponentially here. It's better to mutate instead:

```ts
arr.reduce((obj, item) => ({ ...obj, [item.id]: item }), {}) // slow, ...obj loops over all the keys added on the previous iteration: O(n^2)
arr.reduce((obj, item) => { obj[item.id] = item; return obj; }, {}) // fast, no loops within iteration, O(N)
```

For objects with a small number of entries this might not impact much on performance, but if you tried to reduce something like the tile map (90k+ entries) like this, the difference is seconds vs milliseconds.

Reply:
Great recommendation! I didn't think about this when building it but it should greatly improve the performance of the operations.